Migrated DevHelp docs to github

jcayx opened the pull request.

ayx-orion committed:
initial conversion push

ayx-orion committed:
MD conversion cleanup, syntax highlighting, and missing images

jcayx committed:
initial doc commit

jcayx committed:
Removed html artifacts

jcayx committed:
Fixed linked

jcayx committed:
fixed ayx-python-sdk-v2

jcayx committed:
Changes to cli-and-sdk-faq-v2

jcayx committed:
Removed broken embedded image from html

jcayx committed:
Fixed dcm-input-example-tool

jcayx committed:
more changes to dcm-input-example-tool

jcayx committed:
one more change same file

jcayx committed:
Changed to a .py file to render properly

jcayx committed:
fixed metadata

jcayx committed:
couple more fixes to metadata

jcayx committed:
fixed test-client-command-reference

jcayx committed:
fixed test-client

jcayx committed:
more fixes to test-client

jcayx committed:
playing with code blocks to fix formatting

jcayx committed:
Fixed tabbing again

jcayx committed:
Changed tab again

jcayx committed:
trying to force code block

jcayx committed:
Changed tab again

jcayx committed:
Changed interfering list

jcayx committed:
Trying something else

jcayx committed:
Fixed test-scaffolding

jcayx committed:
added missing docs files

jcayx committed:
Update ayx-plugin-cli-commands.md

jcayx committed:
Update ayx-plugin-cli-overview.md

jcayx committed:
Update ayx-plugin-cli.md

jcayx committed:
Update platform-sdk-quickstart-guide.md

jcayx committed:
file consolidation

jcayx committed:
Merge branch 'devhelp-docs' of https://github.com/alteryx/ayx-developer-sdk into devhelp-docs

jcayx committed:
Merge remote-tracking branch 'remotes/origin/doc-migration-md' into devhelp-docs

jcayx committed:
rearranged files, fixed image links

jcayx committed:
added missing images

jcayx committed:
fixed typo

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+Prerequisites
+-------------
+
+To get started with the Alteryx Core SDK, you need a valid installation
+of [Alteryx Designer](https://www.alteryx.com/).
+
+Python version 3.8.5 is required for Alteryx Designer plugins.
```
TW review:

Update to bullet list, new URL for Alteryx Designer, also Q: is Alteryx Core SDK the name we should use? or should it by Alteryx Platform SDK?

- To get started with the Alteryx Core SDK, you need a valid installation of [Alteryx Designer](https://www.alteryx.com/products/alteryx-designer).
- Python version 3.8.5 is required for Alteryx Designer plugins.

jcayx replied:
latest name is 'Alteryx Platform SDK', fixed and made a bulleted list

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+### Install the Package
+
+After you create and activate your virtual environment, pip install the
+Alteryx Core SDK pip package. For more information on pip, visit the
```
TW review: same as above - Do we still call it Core SDK? Platform SDK? Python SDK? not sure which one we want to go with. @ayx-david-ung

jcayx replied:
'Alteryx Platform SDK' is what we should be calling it

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+
+The `create-ayx-plugin` command takes these parameters:
+
+ -   `--tool-name`: (Required) The name of the new Plugin tool
```
TW review: add a period:

(Required) The name of the new Plugin tool.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+ -   `--tool-name`: (Required) The name of the new Plugin tool
+ -   `--tool-type`: The type of tool to create.
+
+For this example we\'ll name our new tool `MyFirstTool`. We\'ll use the
+tool-type `single-input-single-output` parameter to create the new tool
+with default input and output anchors.
```
TW review:

For this example, name your new tool `MyFirstTool`. Use the
tool-type `single-input-single-output` parameter to create the new tool
with default input and output anchors.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+Now that the Alteryx Python SDK is installed in your virtual environment
+packages, we\'ll create a new [Plugin Tool
+Workspace](https://extensibility.pages.git.alteryx.com/ayx-sdks/getting_started.html#glossary)
+with default files ready to go.
```
TW review:

Now that the Alteryx Python SDK is installed in your virtual environment
packages, create a new [Plugin Tool
Workspace](https://extensibility.pages.git.alteryx.com/ayx-sdks/getting_started.html#glossary)
with default files ready to go.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+
+### Workspace Configuration
+
+When a workspace is created, a `ayx_workspace.json` file is generated
+inside of your workspace directory. This file tracks metadata related to
+your workspace. By default, you do not need to worry about this file
+because it is managed and updated by the CLI commands.
```
TW review:

When a workspace is created, an `ayx_workspace.json` file is generated
inside of your workspace directory. This file tracks metadata related to
your workspace. By default, you don't need to worry about this file
because the CLI commands manage and update it.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+After you have implemented and tested your new Ayx Plugin Tool, you
+might want to distribute it to others. In order to do this, you must
+create a YXI installer. The Core SDK CLI provides a command for
+packaging your workspace and tools into an installer.
```
TW review:

After you implement and test your new Ayx Plugin Tool, you
might want to distribute it to others. In order to do this, you must
create a YXI installer. The Core SDK CLI provides a command to
package your workspace and tools into an installer.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+Run the `create-yxi` command to generate a YXI installer that is ready
+for distribution.
+
+By default, the name of the YXI produced is based on the Ayx Plugin
+Workspace directory name, but you can override the default with the
+`--yxi-name` flag. Alternatively, you can modify the `yxi_name` field in
+the `ayx_workspace.json` file. (See the [Workspace
+Configurations](https://extensibility.pages.git.alteryx.com/ayx-sdks/python-sdk/getting_started.html#workspace-configurations)
+section for details).
```
TW review:

By default, the name of the YXI is based on the Ayx Plugin
Workspace directory name, but you can override the default with the
`--yxi-name` flag. Alternatively, you can modify the `yxi_name` field in
the `ayx_workspace.json` file. (Refer to the [Workspace
Configurations](https://extensibility.pages.git.alteryx.com/ayx-sdks/python-sdk/getting_started.html#workspace-configurations)
section for details).

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+folder. These folders have all the files necessary for Alteryx Designer
+to integrate with your plugin.
+
+Next we\'re going to build your new Ayx Plugin Tool into Designer, where
+you can drag it onto the canvas and connect it to other Designer tools.
+To build `MyFirstTool` into the Designer application, use the
+`designer-install` command.
```
TW review:

Next, you can build your new Ayx Plugin Tool into Designer, where
you can drag it onto the canvas and connect it to other Designer tools.
To build `MyFirstTool` into the Designer application, use the
`designer-install` command.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+
+`ayx_plugin_cli designer-install`
+
+**Congratulations!**
```
TW review: remove this line.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+
+### Ayx Plugin Tool Execution in Designer
+
+When Designer runs a tool, it must look for an engine to use. In the
+case of our MyFirstTool Python tool, the engine is itself the *Python
+interpreter*. The interpreter is built out of Anaconda and includes all
+of the packages indicated in `requirements.txt`.
```
TW review:

When Designer runs a tool, it must look for an engine to use. In the
case of the MyFirstTool Python tool, the engine is itself the *Python
interpreter*. The interpreter is built out of Anaconda and includes all
of the packages indicated in `requirements.txt`.

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+can be changed at any time to any Python file as long as it contains a
+`Plugin` class that registers with the Alteryx Core SDK.
+
+-   For a detailed explanation of the Config.xml file, see [Tool
+    Configurations](https://extensibility.pages.git.alteryx.com/ayx-sdks/python-sdk/config_xml.html).
+-   For information on the `main.py` file within your Ayx Plugin Tool,
+    see [Plugin Code
+    Overview](https://extensibility.pages.git.alteryx.com/ayx-sdks/python-sdk/plugin_code.html).
```
TW review:

For a detailed explanation of the Config.xml file, go to [Tool
    Configurations](https://extensibility.pages.git.alteryx.com/ayx-sdks/python-sdk/config_xml.html).
-   For information on the `main.py` file within your Ayx Plugin Tool,
    go to [Plugin Code
    Overview](https://extensibility.pages.git.alteryx.com/ayx-sdks/python-sdk/plugin_code.html).

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+    see [Plugin Code
+    Overview](https://extensibility.pages.git.alteryx.com/ayx-sdks/python-sdk/plugin_code.html).
+
+### The Configuration Panel GUI
```
TW review:

Remove "The"

ayx-david commented on docs/howto/getting_started_markdown.md:
```diff
@@ -0,0 +1,173 @@
+
+### The Configuration Panel GUI
+
+The Alteryx Core SDK examples provide basic Configuration Panel GUIs,
+however, the Core SDK *only* executes the engine (the Python side of the
+code). Therefore this Getting Started Guide does not cover Alteryx UI
+development. For more information on Configuration Panel GUI
+development, see [Alteryx UI
+SDK](https://help.alteryx.com/developer-help/ayx-ui-sdk).
```
TW review:

The Alteryx Core SDK examples provide basic Configuration Panel GUIs,
however, the Core SDK *only* executes the engine (the Python side of the
code). Therefore this Getting Started Guide doesn't cover Alteryx UI
development. For more information on Configuration Panel GUI
development, go to [Alteryx UI
SDK](https://help.alteryx.com/developer-help/ayx-ui-sdk).

ayx-david commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,196 @@
+
+#### Python SDK v2
+
+Version 2 of AYX Python SDK is now available. For the best experience,
+and to ensure compatibility with the most recent versions of Designer,
+please use [v2 of AYX Python
+SDK](https://help.alteryx.com/developer-help/ayx-python-sdk-v2)
```
TW review: add a period.

Version 2 of AYX Python SDK is now available. For the best experience,
and to ensure compatibility with the most recent versions of Designer,
please use [v2 of AYX Python
SDK](https://help.alteryx.com/developer-help/ayx-python-sdk-v2).

ayx-david commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,196 @@
+location of your workspace and view the files that were created for the
+new plugin.
+
+For more information about the CLI, please visit the [reference
+documents](https://help.alteryx.com/developer-help/ayx-plugin-cli-overview)
```
TW review: add a period:

For more information about the CLI, please visit the [reference
documents](https://help.alteryx.com/developer-help/ayx-plugin-cli-overview).

ayx-david commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,196 @@
+    off a tool. These are also the values that are available to the
+    engine (back end) of your tool.
+
+For the real-time Dev Harness that assists in developing a tool UI,
+check
+out [this](https://github.com/alteryx/dev-harness)
```
TW review:

For the real-time Dev Harness that assists in tool UI development, refer to [Dev Harness](https://github.com/alteryx/dev-harness).

ayx-david commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,196 @@
+    root workspace director. This command builds and installs your
+    plugin into Designer.\
+    \
+    `ayx_plugin_cli designer-install`
```
TW review: typo

director -> directory

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+`conda activate SDKEnv`
+
+This creates an isolated development environment that minimizes the risk
+of creating a package that Alteryx Designer and other users won\'t be
+able to use. It does this by keeping the dependencies required by
+different projects separate.
```
TW review:

This creates an isolated development environment that minimizes the risk
of creating a package that Alteryx Designer and other users won\'t be
able to use. It keeps the dependencies that different projects require separate.

ayx-david-ung replied:
TODO: Next week in new MR @jcayx

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
```
TW review:

Note: If a parameter is not passed in as part of the CLI call, the CLI
waits for standard input for the ignored parameters, even if they're
optional.

ayx-david-ung replied:
TODO: Next week in new MR @jcayx

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: python
```
TW review: capitalize python

Permissible Values: Python

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
```
TW review:

Note: If a parameter is not passed in as part of the CLI call, the CLI
waits for standard input for the ignored parameters, even if they're
optional.

ayx-david-ung replied:
TODO: Next week in new MR @jcayx

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+#### \--tool-name
+
+This parameter becomes the name of the tool. Spaces are allowed but will
+be cleaned into underscores and Pascal case. For example, for a Python
+plugin named "A Test Tool", the filename is `a_test_tool.py`, and the
+`class/ui/config` name is `ATestTool`.
```
TW review:

This parameter becomes the name of the tool. Spaces are allowed but are cleaned into underscores and Pascal case. For example, for a Python
plugin named "A Test Tool", the filename is `a_test_tool.py`, and the
`class/ui/config` name is `ATestTool`.

ayx-david-ung replied:
TODO: Next week in new MR @jcayx

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+This command takes a YXI (any CLI-built or legacy SDK YXI) and installs
+it into Designer.
+
+Note: This command can be run from anywhere, not just Alteryx Workspace
+directories.
```
TW review:

Note: You can run this command from anywhere, not just Alteryx Workspace
directories.

jcayx replied:
changed

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
```
TW review:

Note: If a parameter is not passed in as part of the CLI call, the CLI
waits for standard input for the ignored parameters, even if they're
optional.

jcayx replied:
fixed

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
```
TW review:

Note: If a parameter is not passed in as part of the CLI call, the CLI
waits for standard input for the ignored parameters, even if they're
optional.

jcayx replied:
fixed

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+#### Initialize Workspace
+
+To initialize the plugin workspace, run `sdk-workspace-init`. You will
+be asked to define these items:
```
TW review:

To initialize the plugin workspace, run `sdk-workspace-init`. You need to define these items:

jcayx replied:
fixed

ayx-david commented on docs/references/ayx-plugin-cli.md:
```diff
@@ -0,0 +1,487 @@
+
+#### Add Plugin
+
+Once your workspace is ready, run `create-ayx-plugin`. You will be asked
+to define these items:
```
TW review:

Once your workspace is ready, run `create-ayx-plugin`. You need
to define these items:

jcayx replied:
fixed

ayx-david commented on docs/references/config_xml_markdown.md:
```diff
@@ -0,0 +1,85 @@
+
+ -   `<InputConnections>`: The input anchors for the Ayx Plugin Tool.
+
+      -   `<Connection>`: An individual anchor attached to the tool.
+      -   `<Connection>`: An individual anchor attached to the tool.
+      -   `Name`: The name of the input anchor. This name is
+      -   `Name`: The name of the input anchor. This name is
+          referenced in the Alteryx Core SDK to look up anchors for
+          referenced in the Alteryx Core SDK to look up anchors for
+          receiving and setting data.
+          receiving and setting data.
+      -   `AllowMultiple`: A flag that indicates whether multiple
+      -   `AllowMultiple`: A flag that indicates whether multiple
+          connections can be fed into this anchor.
```
TW review: Looks like some duplicated entries here.

jcayx replied:
fixed

ayx-david commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,67 @@
+Getting Started
+---------------
+
+Getting started in the old SDK typically means looking at an example
+plugin, copying the code and file structure, and then modifying it to
+suit your own needs. The new SDK provides a command-line interface (CLI)
+that takes care of all of this project setup. Access the documentation
+[here](https://help.alteryx.com/developer-help/ayx-plugin-cli-quickstart).
```
TW review:

Getting started in the old SDK typically means looking at an example
plugin, copying the code and file structure, and then modifying it to
suit your own needs. The new SDK provides a command-line interface (CLI)
that takes care of all of this project setup. Go to the [documentation](https://help.alteryx.com/developer-help/ayx-plugin-cli-quickstart) for more information.

jcayx replied:
fixed

ayx-david commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,67 @@
+Development
+-----------
+
+In the original Python SDK, the backend of a tool is developed via a
+class definition that satisfies the interface described in the [AYX
+Plugin Python
+Class](https://help.alteryx.com/current/developer-help/ayxplugin-python-class)
+article.
```
TW review:

In the original Python SDK, the back end of a tool is developed via a
class definition that satisfies the interface described in the [AYX
Plugin Python
Class](https://help.alteryx.com/current/developer-help/ayxplugin-python-class)
article.

jcayx replied:
fixed

ayx-david commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,67 @@
+To alleviate this problem, the new SDK simplifies the interface that
+must be satisfied, to a bare minimum set of requirements.
+
+Similar to the original SDK, in the new SDK a developer must write a
+`Plugin` class. In the new SDK, a base class definition of `Plugin` is
+defined to be used as a parent. This gives the developer a level of
+comfort that they have implemented all necessary methods to have a valid
+Alteryx Designer plugin.
```
TW review:

Similar to the original SDK, in the new SDK, you must write a
`Plugin` class. In the new SDK, a base class definition of `Plugin` is
defined to be used as a parent. This gives you a level of
comfort that you have implemented all necessary methods to have a valid
Alteryx Designer plugin.

jcayx replied:
fixed

ayx-david commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,67 @@
+comfort that they have implemented all necessary methods to have a valid
+Alteryx Designer plugin.
+
+Additionally, in the original SDK, a class called `IncomingInterface`
+was required. This requirement was removed in the new SDK, as incoming
+interfaces and connections are handled behind the scenes by the SDK, and
+made available to the developer via the new `Provider` concept.
```
TW review:

Additionally, in the original SDK, a class called `IncomingInterface`
was required. This requirement was removed in the new SDK, as incoming
interfaces and connections are handled behind the scenes by the SDK, and
made available to you via the new `Provider` concept.

jcayx replied:
fixed

ayx-david commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,67 @@
+Deployment
+----------
+
+Once plugin development is complete, the plugin is often distributed via
+the YXI file format. The instructions for packaging a YXI in the
+original SDK [can be found
+here](https://help.alteryx.com/current/developer-help/package-tool).
+However, this packaging process was significantly simplified by the new
+SDK CLI, described in the [Getting
+Started](https://extensibility.pages.git.alteryx.com/ayx-sdks/getting_started.html)
+guide.
```
TW review:

Once plugin development is complete, the plugin is often distributed via
the YXI file format. You can find the instructions for packaging a YXI in the
original SDK at [Package a Tool](https://help.alteryx.com/current/developer-help/package-tool).
This packaging process was significantly simplified by the new
SDK CLI, described in the [Getting
Started](https://extensibility.pages.git.alteryx.com/ayx-sdks/getting_started.html)
guide.

jcayx replied:
fixed

ayx-david commented on docs/references/faq.md:
```diff
@@ -135,3 +135,147 @@ out_batch = pa.Table.from_arrays([new_data, batch.column('ColumnToKeepAsIs')], s
+
+### _"Do I need to whitelist anything to use the CLI?"_
+
+To download and install the CLI, be notified of a newer version of the
+CLI, or check for updates to the CLI, you need to whitelist
+[https://pypi.org/](https://pypi.org/).
+Otherwise, feel free to use the CLI completely offline.
```
TW review:

To download and install the CLI, receive notifications of newer versions of the
CLI, or check for updates to the CLI, you need to whitelist
[https://pypi.org/](https://pypi.org/).
Otherwise, feel free to use the CLI completely offline.

jcayx replied:
fixed

ayx-david commented on docs/references/faq.md:
```diff
@@ -135,3 +135,147 @@ out_batch = pa.Table.from_arrays([new_data, batch.column('ColumnToKeepAsIs')], s
+
+## _"What's the difference between sdk-workspace-init and create-ayx-plugin commands?"_
+
+-   `sdk-workspace-init`: This command initializes a workspace directory
+    --- think of this as a project playground where all your custom
+    plugins exist. The workspace directory will be setup with a
+    pre-defined folder structure which enables the rest of the CLI
+    commands to properly function.
```
TW review:

`sdk-workspace-init`: This command initializes a workspace directory
    --- think of this as a project playground where all your custom
    plugins exist. The workspace directory is set up with a
    pre-defined folder structure, which enables the rest of the CLI
    commands to properly function.

jcayx replied:
fixed

ayx-david commented on docs/references/file_provider_markdown.md:
```diff
@@ -0,0 +1,272 @@
+Running the File Provider
+-------------------------
+
+The file provider is run from a command-line argument. The only argument
+the user must pass in is a path to a JSON file with this format:
```
TW review:

The file provider is run from a command-line argument. The only argument
you must pass in, is a path to a JSON file with this format:

jcayx replied:
fixed

ayx-david commented on docs/references/file_provider_markdown.md:
```diff
@@ -0,0 +1,272 @@
+3.  `workflow\_config`: This is a string that corresponds to the
+    absolute path of the workflow\'s configuration file, in other words,
+    `path/to/ExampleWorkflowConfig.xml`.
+4.  `inputs`: This is a list of dictionaries that contain information
+    about the tool\'s input connections. It is optional since not all
+    tools have input. The user should specify input information for each
+    input connection that is associated with this tool. Each input
+    dictionary has 3 keys: `anchor_name`, `records`, and `metadata`.
```
TW review:

`inputs`: This is a list of dictionaries that contain information
    about the tool\'s input connections. It's optional since not all
    tools have input. You should specify input information for each
    input connection that is associated with this tool. Each input
    dictionary has 3 keys: `anchor_name`, `records`, and `metadata`.

jcayx replied:
fixed

ayx-david commented on docs/references/file_provider_markdown.md:
```diff
@@ -0,0 +1,272 @@
+         path of the input connection\'s metadata file, in other words,
+         `{"anchor_name": "AnchorName", "records": "path/to/InputRecords.csv", "metadata": "path/to/InputMetadata.xml"}`.
+
+5.  `outputs`: This is a list of dictionaries that specify where the
+    tool\'s output anchor information should be stored. It is optional
+    since not all tools have output. The user should specify output
+    information for each output anchor that is associated with this
+    tool. Each output dictionary has 3 keys: `anchor_name`, `records`,
+    and `metadata`.
```
TW review:

`outputs`: This is a list of dictionaries that specify where the
    tool\'s output anchor information should be stored. It's optional
    since not all tools have output. You should specify output
    information for each output anchor that is associated with this
    tool. Each output dictionary has 3 keys: `anchor_name`, `records`,
    and `metadata`.

jcayx replied:
fixed

ayx-david commented on docs/references/metadata.md:
```diff
@@ -0,0 +1,274 @@
+
+### on_complete Method
+
+On input plugins, a complete schema created with metadata can be
+exported to Designer for a precise data schema.
```
TW review:

On input plugins, you can export a complete schema created with metadata to Designer for a precise data schema.

jcayx replied:
fixed

ayx-david commented on docs/references/metadata.md:
```diff
@@ -0,0 +1,274 @@
+
+### Define a Schema at \_\_init\_\_
+
+In this example, we specify type `int16` for
+column `volts` and type
+`string` for column
+`device`.
```
TW review:

This example specifies type `int16` for
column `volts` and type
`string` for column
`device`.

jcayx replied:
fixed

ayx-david commented on docs/references/metadata.md:
```diff
@@ -0,0 +1,274 @@
+    }
+
+###### Note
+You can see in the above example that a number represents the type.
+Please refer to the Types section below for the mapping between type id
+and type names.
```
TW review: change id to ID

jcayx replied:
fixed

ayx-david commented on docs/references/platform-sdk.md:
```diff
@@ -0,0 +1,94 @@
+The Platform SDK lets you extend your Alteryx experience by building
+custom tools and plugins.
+
+While Alteryx Designer provides a wide range of functionality with the
+available tools, you might find that you perform a specific action or
+task that could be better served by creating a custom tool or plugin.
+You can distribute custom tools within your organization and environment
+and still leverage the speed and flexibility of Designer and Gallery.
```
TW review: update Gallery -> Server

jcayx replied:
fixed

ayx-david commented on docs/references/plugin_code_markdown.md:
```diff
@@ -0,0 +1,83 @@
+The `on_record_batch` method is called for each input connection on an
+anchor.
+
+This method also receives an `Anchor` object (as well as all the records
+and data sent from E2) in the form of a [PyArrows
+Table](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html).
```
TW review: change E2 to AMP

jcayx replied:
fixed

ayx-david commented on docs/references/plugin_code_markdown.md:
```diff
@@ -0,0 +1,83 @@
+and data sent from E2) in the form of a [PyArrows
+Table](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html).
+
+In this method, the plugin writer can manipulate the data before writing
+the data to the output anchor using
+`self.provider.write_to_anchor(self.output_anchor_name, table)`
```
TW review:

In this method, you can manipulate the data before writing
the data to the output anchor using
`self.provider.write_to_anchor(self.output_anchor_name, table)`

jcayx replied:
fixed

ayx-david commented on docs/references/plugin_code_markdown.md:
```diff
@@ -0,0 +1,83 @@
+
+The `on_complete` method is called at the end of the runtime execution.
+
+This typically does any cleanup required for the Ayx Plugin Tool. If the
+plugin is an Input tool-type, this method is used to read in the data
+from the datasource and push it to the output anchor. (since an Input
+tool-type has no input anchors or connections, and therefore
+`on_incoming_connection_complete` and `on_record_batch` are not called).
```
TW review:

This method typically does any cleanup required for the Ayx Plugin Tool. If the
plugin is an Input tool type, this method is used to read the data
from the data source and push it to the output anchor (since an Input
tool-type has no input anchors or connections, and therefore
`on_incoming_connection_complete` and `on_record_batch` are not called).

jcayx replied:
fixed

ayx-david commented on docs/references/plugin_code_markdown.md:
```diff
@@ -0,0 +1,83 @@
+Proxy Environment Configuration
+-------------------------------
+
+If you are running your plugin on a network that requires a proxy, you
+might encounter errors when you make external requests (An API as a data
+source, for example) We provide a convience function to inherit
+configuration from Designer in `proxy_requests`. The
+`proxy_requests.create_proxied_session` function returns a `PACSession`
+initialized with any found Designer proxy authentication settings. If
+you need any addition configuration, you can extend the `PACSession`
+further; see the [PyPAC Docs](https://pypac.readthedocs.io/en/latest/)
+for more info.
```
TW review:

If you run your plugin on a network that requires a proxy, you
might encounter errors when you make external requests (an API as a data
source, for example). We provide a convenient function to inherit
configuration from Designer in `proxy_requests`. The
`proxy_requests.create_proxied_session` function returns a `PACSession`
initialized with any found Designer proxy authentication settings. If
you need any additional configuration, you can extend the `PACSession`
further. Refer to the [PyPAC Docs](https://pypac.readthedocs.io/en/latest/)
for more info.

jcayx replied:
fixed

ayx-david commented on docs/references/plugin_lifecycle_markdown.md:
```diff
@@ -0,0 +1,88 @@
+plugin. Any resources required to read from and write to Designer can be
+obtained via methods/properties on the `ProviderV2`.
+
+In Alteryx Designer, input anchors can have 1 or many connections. If
+you have any input connections, in the next stage of the lifecycle
+consists of each of those input connections sends data downstream. From
```
TW review:

In Alteryx Designer, input anchors can have 1 or many connections. If
you have any input connections, the next stage of the lifecycle
consists of each of those input connections sending data downstream. From

jcayx replied:
fixed

ayx-david commented on docs/references/plugin_lifecycle_markdown.md:
```diff
@@ -0,0 +1,88 @@
+`AMPProviderV2`, which sets incoming and outgoing anchor information,
+including the name and metadata. Additionally, since `__init__` runs in
+update-only mode, you should aim to perform a minimal amount of
+processing in this methods, since a fast update makes users happy.
```
TW review:

Since the goal of an update-only run is to build output metadata, the
`__init__` method is run. This method has an instance of the
`AMPProviderV2`, which sets incoming and outgoing anchor information,
including the name and metadata. Additionally, since `__init__` runs in
update-only mode, you should aim to perform a minimal amount of
processing in this method, since a fast update makes users happy.

jcayx replied:
fixed

ayx-david commented on docs/references/test-client.md:
```diff
@@ -0,0 +1,373 @@
+
+#### AYX SDK CLI
+
+The `ayx-sdk-cli` is the newest addition to the Alteryx Developer
+Experience product suite. It currently contains the Test Client and is
+the next iteration of CLI. The `ayx-sdk-cli` will house all the
+functionality for Desktop and future Cloud Developer Experience
+offerings.
```
TW review:

The `ayx-sdk-cli` is the newest addition to the Alteryx Developer
Experience product suite. It currently contains the Test Client and is
the next iteration of the CLI. The `ayx-sdk-cli` houses all the
functionality for Desktop and future Cloud Developer Experience
offerings.

jcayx replied:
fixed

ayx-david commented on docs/references/third_party_packages_markdown.md:
```diff
@@ -0,0 +1,86 @@
+1.  Manually add dependencies. If you use this approach, make sure to
+    include all dependencies (including any sub-dependencies).
+
+2. Use `pip freeze > requirements.txt` to generate a new
+requirements.txt file in the workspace. You will need to prune this file
+to remove any dependencies that aren\'t explicitly imported or required
+by your plugin. Do this before you copy or overwrite the contents to the
+`requirements-thirdparty.txt` file
```
TW review:

2. Use `pip freeze > requirements.txt` to generate a new
requirements.txt file in the workspace. You need to prune this file
to remove any dependencies that aren't explicitly imported or required
by your plugin. Do this before you copy or overwrite the contents to the
`requirements-thirdparty.txt` file.

jcayx replied:
fixed

ayx-david commented on docs/references/virtual_environments_markdown.md:
```diff
@@ -0,0 +1,47 @@
+Developing in a Virtual Environment
+-----------------------------------
+
+We recommend that you develop your custom plugins inside a virtual
+environment. This creates an isolated development environment that
+minimizes the risk of creating a package that Alteryx Designer and other
+users won\'t be able to use. It does this by keeping the dependencies
+required by different projects separate.
```
TW review:

We recommend that you develop your custom plugins inside a virtual
environment. This creates an isolated development environment that
minimizes the risk of creating a package that Alteryx Designer and other
users won't be able to use. To do this, it keeps the dependencies
required by different projects separate.

jcayx replied:
fixed

jcayx committed:
TW requested changes

jcayx committed:
more TW changes

ayx-david-ung commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,194 @@
+Version 2 of AYX Python SDK is now available. For the best experience,
+and to ensure compatibility with the most recent versions of Designer,
+please use v2 of [AYX Python
+SDK.](https://help.alteryx.com/developer-help/ayx-python-sdk-v2)
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,194 @@
+
+-   To learn more about v2 of the AYX Python SDK, go to [Changes and
+    Improvements to Python SDK in
+    v2.0.0](https://help.alteryx.com/developer-help/changes-and-improvements-python-sdk-v2).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,194 @@
+The steps below illustrate how to use the latest developer tools to
+create a custom plugin or tool for use in Alteryx. For more information
+about new SDKs and their usage within Alteryx, please visit [Platform
+SDK](https://help.alteryx.com/developer-help/platform-sdk).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,194 @@
+new plugin.
+
+For more information about the CLI, please visit the [reference
+documents](https://help.alteryx.com/developer-help/ayx-plugin-cli-overview).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,194 @@
+2021.2 and 2021.3. For the best experience, and to ensure compatibility
+with the most recent versions of Designer, please use [version 2 of AYX
+Python
+SDK](https://help.alteryx.com/developer-help/ayx-python-sdk-v2).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/howto/platform-sdk-quickstart-guide.md:
```diff
@@ -0,0 +1,194 @@
+view the changes.
+
+For more information about the Python SDK, please go to the [reference
+documents](https://help.alteryx.com/developer-help/ayx-python-sdk) or access the full [Python
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/ayx-python-sdk-v2-example-tools.md:
```diff
@@ -0,0 +1,108 @@
+### Example Tools
+
+These are example tools created using version 2 the [AYX Python
+SDK](https://help.alteryx.com/developer-help/ayx-python-sdk). When the tools are installed, they display in the SDK
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,65 @@
+---------
+
+The original [Alteryx Python
+SDK](https://help.alteryx.com/current/developer-help/python-engine-sdk)
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,65 @@
+Getting started in the old SDK typically means looking at an example
+plugin, copying the code and file structure, and then modifying it to
+suit your own needs. The new SDK provides a command-line interface (CLI)
+that takes care of all of this project setup. Go to the [documentation](https://help.alteryx.com/developer-help/ayx-plugin-cli-quickstart) for more information.
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/differences_with_original_sdk_markdown.md:
```diff
@@ -0,0 +1,65 @@
+In the original Python SDK, the back end of a tool is developed via a
+class definition that satisfies the interface described in the [AYX
+Plugin Python
+Class](https://help.alteryx.com/current/developer-help/ayxplugin-python-class)
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/faq.md:
```diff
@@ -135,3 +135,147 @@ out_batch = pa.Table.from_arrays([new_data, batch.column('ColumnToKeepAsIs')], s
+
+The original [Python Engine
+SDK](https://help.alteryx.com/developer-help/python-engine-sdk) is complex, runs in-process, and can be slow. The [AYX
+Python SDK](https://help.alteryx.com/developer-help/ayx-python-sdk) design solves these issues. It runs
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/faq.md:
```diff
@@ -135,3 +135,147 @@ out_batch = pa.Table.from_arrays([new_data, batch.column('ColumnToKeepAsIs')], s
+
+## _"Python Engine SDK and AYX Python SDK. What's the difference?"_
+
+[AYX Python SDK](https://help.alteryx.com/developer-help/ayx-python-sdk-v2) is a new SDK product. It improves upon
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/metadata.md:
```diff
@@ -0,0 +1,273 @@
+![Workflow Metadata Lifecycle Diagram](assets/workflow-metadata-lifecycle.png)
+
+You can find a more complete example at [Metadata Plugin
+Example](https://help.alteryx.com/developer-help/metadata-plugin-example).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/platform-sdk.md:
```diff
@@ -0,0 +1,94 @@
+2.  The engine, which processes logic to handle records passed to and
+    from the Alteryx Engine. To create the engine component, use the
+    [AYX Python
+    SDK](https://help.alteryx.com/developer-help/ayx-python-sdk).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/platform-sdk.md:
```diff
@@ -0,0 +1,94 @@
+
+To help facilitate the creation of a custom tool or plugin, you can
+leverage the [AYX Plugin
+CLI](https://help.alteryx.com/developer-help/ayx-plugin-cli).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/platform-sdk.md:
```diff
@@ -0,0 +1,94 @@
+### SDK Quickstart Guide
+
+Visit the [Platform SDK Quickstart
+Guide](https://help.alteryx.com/developer-help/platform-sdk-quickstart-guide) to learn how to use the latest developer tools to
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/platform-sdk.md:
```diff
@@ -0,0 +1,94 @@
+
+Visit these additional SDK resources:
+
+-   [SDK Quickstart
```
Remove. Delete this entire section. Anything under "Explore SDK Resources"

Lines 67 - 94

ayx-david-ung commented on docs/references/test-client.md:
```diff
@@ -0,0 +1,373 @@
+
+
+## Command Reference 
+Go to [Test Client Command Reference](https://help.alteryx.com/developer-help/test-client-command-reference) for detailed information.
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung commented on docs/references/test-client.md:
```diff
@@ -0,0 +1,373 @@
+`ayx-sdk-cli.exe plugin run [WORKSPACE_PATH]::[PLUGIN_NAME] --input <[SOURCE][:TYPE[:OPTIONS]][@ANCHOR]> --output <[DESTINATION][:TYPE[:OPTIONS]][@ANCHOR]> --runtime-command`
+
+For more information, go to [Test
+Client](https://help.alteryx.com/developer-help/test-client).
```
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version

ayx-david-ung reviewed (changes requested):
Extra clean up required.

WE have a few links that are still pointing to the old content. Point them to the new references on ayx-developer-sdk repo (the new MD files). They should point to the corresponding markdown version

ayx-david-ung committed:
Updating URL and Deleting old content

Updating URL and Deleting old content

~30 links that are pointing to our old content.

Point there to the new reference on ayx-developer-sdk repo. This should point to the corresponding markdown version.

ayx-david-ung committed:
Updating Guides and Removing old Docs

We do not need File Provider.

The Guides we pointing to bad links

ayx-david-ung reviewed (approved):
LGTM
